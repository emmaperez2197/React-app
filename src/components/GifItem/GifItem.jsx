
const GifItem = ({title, url}) => {

    return ( 
        <di className='card'>
            <img src={url} alt={title}></img>
            <p>{title}</p>

        </di>
     );
}
 
export default  GifItem;