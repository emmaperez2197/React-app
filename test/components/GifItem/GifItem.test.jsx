import { render, screen} from '@testing-library/react'
import GifItem from '../../../src/components/GifItem'


describe('Test component <GifItem/>', ()=>{

    const title = 'Naruto'
    const url = 'https://naruto-uzumaki/images.jpg'

    test('debe de hacer match con el snapshot', ()=>{
        //con el render renderizamosel componente desde nuestros test
        //container es una propiedad de render para poder usar el snapshot (toMatchSnapshot)
        const {container} = render(<GifItem title={title} url={url}/>)

        expect(container).toMatchSnapshot()
    })

    test('debe devolver la URL y el ALT indicado ', ()=>{

        render(<GifItem title={title} url={url} />)
        
        //con el scree.getByrol traemos la etiqueta 'img' y luego corroboramos sus atributos url y alt
        // expect(screen.getByRole('img').src).toBe(url)

        //Forma mas optima


        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(alt);


    })

    test('Debe devolver el titulo del componente',() =>{

        render(<GifItem title={title} url={url} />)

        //screen.getByText(title) chequea que este el valor del texto
        expect(screen.getByText(title)).toBeTruthy();
    })
})