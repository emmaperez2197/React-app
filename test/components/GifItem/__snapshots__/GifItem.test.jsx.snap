// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto-uzumaki/images.jpg"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
